api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerUnitTest.java

```diff
@@ -124,6 +147,132 @@ public void testAttachDiskToValidRuns() {
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyCollections() {
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(new ArrayList<>(), new HashMap<>());
+        assertTrue(CollectionUtils.isEmpty(actualParameters));
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyEnvVars() {
+        final List<PipelineRunParameter> parameters = new ArrayList<>();
+        parameters.add(new PipelineRunParameter(PARAM_NAME_1, TEST_STRING));
+
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(parameters, new HashMap<>());
+        assertEquals(parameters, actualParameters);
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithEmptyParams() {
+        final Map<String, String> envVars = new HashMap<>();
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
+
+        final List<PipelineRunParameter> actualParameters =
+                pipelineRunManager.replaceParametersWithEnvVars(new ArrayList<>(), envVars);
+        assertTrue(CollectionUtils.isEmpty(actualParameters));
+    }
+
+    @Test
+    public void shouldNotReplaceEnvVarsIfNoNeeded() {
+        final List<PipelineRunParameter> parameters = new ArrayList<>();
+        parameters.add(new PipelineRunParameter(PARAM_NAME_1, TEST_STRING));
+        final Map<String, String> envVars = new HashMap<>();
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
+
+        final List<PipelineRunParameter> actualParameters
+                = pipelineRunManager.replaceParametersWithEnvVars(parameters, envVars);
+        assertEquals(TEST_STRING, actualParameters.get(0).getValue());
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithTestEnv() {
+        testEnvVarsReplacement("test/${%s}", "test/%s");
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithTestEnvAtTheEndOfTheLine() {
+        testEnvVarsReplacement("test/$%s", "test/%s");
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithTestEnvAtTheMiddleOfTheLine() {
+        testEnvVarsReplacement("test/$%s/", "test/%s/");
+    }
+
+    @Test
+    public void shouldReplaceEnvVarsWithTestEnvWithSeveralVariables() {
+        final Map<String, String> envVars = new HashMap<>();
+        envVars.put(ENV_VAR_NAME, ENV_VAR_VALUE);
+
+        final String paramValue = String.format("test/$%s/${%s}/$%s/", ENV_VAR_NAME, ENV_VAR_NAME, ENV_VAR_NAME);
+        final String expectedValue = String.format("test/%s/%s/%s/", ENV_VAR_VALUE, ENV_VAR_VALUE, ENV_VAR_VALUE);
+        final List<PipelineRunParameter> actualParameters = pipelineRunManager.replaceParametersWithEnvVars(
+                Collections.singletonList(new PipelineRunParameter(PARAM_NAME_1, paramValue)), envVars);
+        checkResolvedValue(actualParameters, paramValue, expectedValue);
+    }
+
+    private void testEnvVarsReplacement(final String paramValuePattern, final String expectedValuePattern) {
```

Comment:
Let's call these method `assertEnvVarsReplacement` similarly to other private methods in the class.

Reply:
Done!